From baf8790073fcc7e2945299280e7dff46573b8d4d Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Mon, 16 Jul 2018 14:43:28 +0800
Subject: [PATCH 4174/5242] MLK-18427-02 ARM64: dts: freescale: add busfreq
 mode for imx8mm

commit  afdfc42a7d9c71c305ad1baca5db1b143a096d4d from
https://source.codeaurora.org/external/imx/linux-imx.git

Add busfreq node support on i.MX8MM. busfreq support is enabled by
default, but it need to be disabled on DDR4 validation board at present.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
(cherry picked from commit 6d5307eb88a6f5bfdb7123fd9165681cd5a31c7d)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mm-ddr4-val.dts     |    4 ++++
 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi      |   18 ++++++++++++++++++
 2 files changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-val.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-val.dts
index bbc33a0..797f21d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-val.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-val.dts
@@ -39,6 +39,10 @@
 			enable-active-high;
 		};
 	};
+
+	busfreq {
+		status = "disabled";
+	};
 };
 
 &iomuxc {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
index a3a3184..b0162f6 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
@@ -110,6 +110,24 @@
 		interrupt-parent = <&gic>;
 	};
 
+	busfreq { /* BUSFREQ */
+		compatible = "fsl,imx_busfreq";
+		clocks = <&clk IMX8MM_DRAM_PLL_OUT>, <&clk IMX8MM_CLK_DRAM_ALT_SRC>,
+			 <&clk IMX8MM_CLK_DRAM_APB_SRC>, <&clk IMX8MM_CLK_DRAM_APB_PRE_DIV>,
+		         <&clk IMX8MM_CLK_DRAM_CORE>, <&clk IMX8MM_CLK_DRAM_ALT_ROOT>,
+			 <&clk IMX8MM_SYS_PLL1_40M>, <&clk IMX8MM_SYS_PLL1_100M>,
+			 <&clk IMX8MM_SYS_PLL2_333M>, <&clk IMX8MM_CLK_NOC_DIV>,
+			 <&clk IMX8MM_CLK_AHB_DIV>, <&clk IMX8MM_CLK_MAIN_AXI_SRC>,
+			 <&clk IMX8MM_CLK_24M>, <&clk IMX8MM_SYS_PLL1_800M>;
+		clock-names = "dram_pll", "dram_alt_src", "dram_apb_src", "dram_apb_pre_div",
+			      "dram_core", "dram_alt_root", "sys_pll1_40m", "sys_pll1_100m",
+			      "sys_pll2_333m", "noc_div", "ahb_div", "main_axi_src", "osc_24m",
+			      "sys_pll1_800m";
+		interrupts = <GIC_SPI 74 IRQ_TYPE_LEVEL_HIGH>, <GIC_SPI 75 IRQ_TYPE_LEVEL_HIGH>,
+			     <GIC_SPI 76 IRQ_TYPE_LEVEL_HIGH>, <GIC_SPI 77 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-name = "irq_busfreq_0", "irq_busfreq_1", "irq_busfreq_2", "irq_busfreq_3";
+	};
+
 	ddr_pmu0: ddr_pmu@3d800000 {
 		compatible = "fsl,imx8m-ddr-pmu", "fsl,imx8-ddr-pmu";
 		reg = <0x0 0x3d800000 0x0 0x400000>;
-- 
1.7.9.5

